File: app/src/main/java/com/example/newsweatherapp/Rest/ApiInterface.java
package com.example.newsweatherapp.Rest;


import com.example.newsweatherapp.Model.NewsListResponseModel;


import retrofit2.Call;

import retrofit2.http.GET;
import retrofit2.http.Query;


public interface ApiInterface {

    @GET("articles/")
    Call<NewsListResponseModel> NewAppResponse(@Query("params") String params);


}

File: app/src/main/java/com/example/newsweatherapp/Rest/ApiInterface.kt
package com.example.newsweatherapp.Rest

import retrofit2.http.GET
import com.example.newsweatherapp.Model.NewsListResponseModel
import retrofit2.Call
import retrofit2.http.Query
import retrofit2.http.QueryMap

interface ApiInterface {

    @GET("articles/")
    fun NewAppResponse(@QueryMap params: Map<String, String>): Call<NewsListResponseModel>
}